prow/gerrit/adapter/trigger.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package adapter
+
+import (
+	"k8s.io/test-infra/prow/pjutil"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/gerrit/client"
+)
+
+const layout = "2006-01-02 15:04:05"
+
+// messageFilter builds a filter for jobs based on the messageBody matching the trigger regex of the jobs.
+func messageFilter(lastUpdate time.Time, change client.ChangeInfo, presubmits []config.Presubmit) (pjutil.Filter, error) {
+	var filters []pjutil.Filter
+	currentRevision := change.Revisions[change.CurrentRevision].Number
+	for _, message := range change.Messages {
+		// Skip comments not germane to this plugin
+		if !pjutil.TestAllRe.MatchString(message.Message) {
```

Comment:
Why not support `/ok-to-test` and `/retest`?

Reply:
Since gerrit doesn't have contexts there's no easy way to tell which past jobs have failed. But, I'm planning to work on this as a follow up.

Reply:
yeah technically this is do-able, we can poll all prowjobs associated with the patchset and see which most recent ones failed. Maybe put a TODO for now.